GameOfRunes/GameOfRunes/GameEngine/Systems/MoveSystem.swift

```diff
@@ -57,16 +58,16 @@ class MoveSystem: GKComponentSystem<MoveComponent>, System {
     }
     
     private func checkEnemyEndPointCollision() {
-        guard let endpointComponent = gameEngine?.entities(for: .player).first as? EndPointEntity,
+        guard let endpointComponent = gameEngine?.entities(for: .player).compactMap({ $0 as? EndPointEntity }).first,
```

Comment:
I guess we can use `entities(for: .endPointEntity)` instead? Since we don't need other `.player` entities